aria_data_loaders/aria_data_utils/aria_streamer.py

```diff
@@ -0,0 +1,1101 @@
+        self.xyz_trajectory = np.empty([len(self.mps_trajectory), 3])
+
+        # Timestamps for mps in seconds
+        self.trajectory_s = np.empty([len(self.mps_trajectory)])
+
+        # Different transformations along the trajectory
+        self.ariaWorld_T_device_trajectory = []  # type: List[Any]
+        self.ariaCorrectedWorld_T_device_trajectory = []  # type: List[Any]
+        self.ariaCorrectedWorld_T_cpf_trajectory = []  # type: List[Any]
+
+        # Setup some generic transforms
+        self.device_T_cpf = sp.SE3(
+            self.device_calib.get_transform_device_cpf().to_matrix()
+        )
+        self.cpf_T_device = self.device_T_cpf.inverse()
+
+        # Initialize Trajectory after setting up cpf transforms
+        self.initialize_trajectory()
+
+        # sensor_calib_list = [device_calib.get_sensor_calib(label) for label in stream_names][0]
+        # Record VRS timestamps of interest based upon user input during vrs parsing
+        self.vrs_idx_of_interest_list = []  # type: List[Any]
+
+    def plot_rgb_and_trajectory(
+        self,
+        marker_pose: sp.SE3,
+        device_pose_list: List[sp.SE3],
+        rgb: np.ndarray,
+        traj_data: np.ndarray = None,
+    ):
+        """
+        Plot RGB image with trajectory
+
+        Args:
+            marker_pose (sp.SE3): Pose of marker in frame of reference
+            device_pose_list (List[sp.SE3]): List of device poses in frame of reference
+            rgb (np.ndarray): RGB image
+            traj_data (np.ndarray): Trajectory data
+        """
+        fig = plt.figure(figsize=plt.figaspect(2.0))
+        fig.suptitle("A tale of 2 subplots")
+
+        _ = fig.add_subplot(1, 2, 1)
+        plt.imshow(rgb)
+
+        scene = Scene()
+        scene.add_frame("correctedWorld", pose=self.device_T_cpf)
+        scene.add_camera(
+            "dock", frame="correctedWorld", pose_in_frame=marker_pose, size=AXES_SCALE
+        )
+        for i in range(len(device_pose_list)):
+            scene.add_camera(
+                f"device_{i}",
+                frame="correctedWorld",
+                pose_in_frame=device_pose_list[i],
+                size=AXES_SCALE,
+            )
+
+        plt_ax = scene.visualize(fig=fig, should_return=True)
+        plt_ax.plot(traj_data[:, 0], traj_data[:, 1], traj_data[:, 2])
+        plt.show()
+
+    def _init_april_tag_detector(self) -> Dict[str, Any]:
+        """
+        Initialize April tag detector object for pose estimation of QR code
+
+        Can only detect dock code
+        """
+        focal_lengths = self._dst_calib_params.get_focal_lengths()  # type:ignore
+        principal_point = self._dst_calib_params.get_principal_point()  # type:ignore
+        calib_dict = {
+            "fx": focal_lengths[0].item(),
+            "fy": focal_lengths[1].item(),
+            "ppx": principal_point[0].item(),
+            "ppy": principal_point[1].item(),
+            "coeffs": [0.0, 0.0, 0.0, 0.0, 0.0],
+        }
```

Comment:
Move this to constructor or somewhere else. Rectification is independent of April tag detection or object detection.

Reply:
This has been updated, as we moved it into a new class.